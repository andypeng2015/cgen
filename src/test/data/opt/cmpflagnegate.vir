module foo

export function w $foo(w %x) {
@start:
  %c1 = lt.w %x, 5_w
  %f1 = flag.w %c1
  %c2 = le.w %f1, 0_w
  %f2 = flag.w %c2
  ret %f2
}
