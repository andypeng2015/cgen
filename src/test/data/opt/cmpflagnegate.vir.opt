module foo

export function w $foo(w %x) {
@2:
  %2 = ge.w %x, 0x5_w ; @9
  %3 = flag.w %2 ; @10
  ret %3
}
