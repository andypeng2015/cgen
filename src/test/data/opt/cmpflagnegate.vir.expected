module foo

export function w $foo(w %x) {
@2:
  %0 = ge.w %x, 0x5_w ; @7
  %1 = flag.w %0 ; @8
  ret %1
}
